src/ProcessTools.cpp

```diff
@@ -57,44 +77,369 @@ SUPPRESS_VC_WARNING(4548) // This was added to suppress a warning in MSVC's impl
+            { Result.ConsoleOutput.append(PipeBuf,static_cast<size_t>(BytesRead)); }
+        ::close(ChildInfo.ChildPipe);
+
+        int Status = -1;
+        ::waitpid(ChildInfo.ChildPID,&Status,0);
+        if( WIFEXITED(Status) ) {
+            Result.ExitCode = WEXITSTATUS(Status);
+        }else if( WIFSIGNALED(Status) ) {
+            Result.ExitCode = WTERMSIG(Status);
+        }else{
+            // No idea what else could have happened
+            Result.ExitCode = Status;
         }
-        RESTORE_WARNING_STATE
+        // Trim newlines
+        while( !Result.ConsoleOutput.empty() &&
+               Result.ConsoleOutput.back() == '\n' )
+        {
+            Result.ConsoleOutput.pop_back();
+        }
+#endif // MEZZ_Windows
+        return Result;
+    }
+}
+
+namespace Mezzanine {
+namespace Testing {
+    ///////////////////////////////////////////////////////////////////////////////
+    // Output to String
 
-    }// Testing
+    CommandResult RunCommand(const StringView ExecutablePath, const StringView Command)
+    {
+        const Mezzanine::String SafePath( SanitizeProcessCommand(ExecutablePath) );
+        const Mezzanine::String SafeCommand( SanitizeProcessCommand(Command) );
+        if( SafeCommand != Command )
+            { throw std::runtime_error("Command included unsafe characters, it would not run correctly."); }
+        return RunCommandImpl(SafePath,SafeCommand);
+    }
+
+    CommandResult RunCommand(const StringView Command)
+    {
+#ifdef MEZZ_Windows
+        // Windows is happy to parse just a single string for everything.
+        StringView Empty;
+        return RunCommand(Empty,Command);
+#else // Mezz_Windows
+        // Posix is NOT happy to do the same.  The strings must be separate.
+        /// @todo Maybe handle filename paths with spaces?
+        size_t SplitPos = Command.find_first_of(" \t");
+        const Mezzanine::String ExecPath{ Command.substr(0,SplitPos) };
+        return RunCommand(ExecPath,Command);
+#endif // MEZZ_Windows
+    }
+
+    String GetCommandOutput(const StringView ExecutablePath, const StringView Command)
+    {
+        const Mezzanine::String SafePath( SanitizeProcessCommand(ExecutablePath) );
+        const Mezzanine::String SafeCommand( SanitizeProcessCommand(Command) );
+        if( SafeCommand != Command )
+            { throw std::runtime_error("Command included unsafe characters, it would not run correctly."); }
+        return RunCommandImpl(SafePath,SafeCommand).ConsoleOutput;
+    }
+
+    String GetCommandOutput(const StringView Command)
+    {
+#ifdef MEZZ_Windows
+        // Windows is happy to parse just a single string for everything.
+        StringView Empty;
+        return GetCommandOutput(Empty,Command);
+#else // Mezz_Windows
+        // Posix is NOT happy to do the same.  The strings must be separate.
+        /// @todo Maybe handle filename paths with spaces?
+        size_t SplitPos = Command.find_first_of(" \t");
```

Comment:
This little snippet is done multiple times, even the todo. Maybe extract to a function and mark that function with one today so a future dev is not bit when they only change one or the other.

Reply:
Done.